libs/pageserver_api/src/upcall_api.rs

```diff
@@ -20,12 +22,20 @@ pub struct ReAttachRequest {
     pub register: Option<NodeRegisterRequest>,
 }
 
-#[derive(Serialize, Deserialize)]
+fn default_mode() -> LocationConfigMode {
+    LocationConfigMode::AttachedSingle
+}
+
+#[derive(Serialize, Deserialize, Debug)]
 pub struct ReAttachResponseTenant {
     pub id: TenantShardId,
-    pub gen: u32,
-}
+    /// Mandatory if LocationConfigMode is None or set to an Attached* mode
+    pub gen: Option<u32>,
```

Comment:
Thanks for reminding about it, created an issue to make generation optional on our side https://github.com/neondatabase/cloud/issues/11469

Currently we return zero generation for secondary location, is it ok for now? generations are not read from re-attach response if `mode=Secondary`?

Reply:
>Currently we return zero generation for secondary location, is it ok for now? generations are not read from re-attach response if mode=Secondary?

That should be fine.